nsapass

```diff
@@ -229,63 +221,59 @@ def genpass(gen, showpass):
             break
     return password
 
-def checkquery(db, username, index, arc=None):
-    """checks if supplied query (username, index) exists in the database
+def checkquery(db, id, rev, old=None):
+    """checks if supplied query (id, rev) exists in the database
     db.  in case it doesnt't, it tries to guesses sensibly (by guessing
     nearest match, or by prompting for correction).  if no sensible guess
     is possible, it prompts the user.  if there is no hope even asking the
     user, it exits right here after leaving a beautiful error message
     behind"""
     if len(db) == 0:
         printerr('databse is empty')
-    usernames = list(db)
-    if arc is not None: # filter users without relevant entries
-        usernames = [u for u in usernames
-            if len([() for user in db[u] if user[KEY_ARC] == arc]) > 0]
-    if username is None:
-        if len(usernames) == 1:
+    ids = list(db)
+    if old == True:  # filter users without old entries
+        ids = [i for i in ids if len(db[i]) > 1]
+    if old == False: # filter users without active entries
+        ids = [i for i in ids if db[i][0] is not None]
```

Comment:
I'm not sure. I imagined that for a user who never deletes old entries, the history can become quite long.

Reply:
so that'll be a feature by users who like to query `db` by to find out which entries are not trimmed to trim them?

i wonder if the best use case is for using this feature in `nsapass list` as follows:

- `nsapass list` to list only latest revisions of entries not deleted (aka not archived).
- `nsapass list -o` to cause the listing to expand the entries by showing their past revisions (then the user can pipe it to `less` to navigate and see which entries he needs to trim).
- `nsapass list -a` to also show the deleted/archived entries, but without showing their past revisions (only latest revision).
- `nsapass list -ao` to do both: all entries and all revisions.

so i wonder if defining `-a` and `-o` as flags that extend/expand `db`'s access scope (by subcommands) makes more sense as opposed to making them as flags that partition the db into (1) part with history, (2) part without history, (3) part archived, (4) part unarchived.

`/me goes to think a bit more`